dispatchr-ruby/spec/controllers/request_items_controller_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+RSpec.describe RequestItemsController, :type => :controller do
+
+  def create_request_item
+    post :create, {request_item: {request_id: 1, max_price: 100, quantity_description: "5", item: { name: "bananas" }}}
+  end
+
+  def create_bad_request_item
+    post :create, {request_item: {max_price: 100, quantity_description: "5", item: { name: "bananas" }}}
+  end
+
+  describe 'GET #show' do
+    context 'when id is valid' do
+      it 'returns 200 status' do
+        create_request_item
+        request_item = RequestItem.last
+        get :show, {id: request_item.id}
+        expect(response).to have_http_status(200)
+      end
+    end
+
+    context 'when id is invalid' do
+      it 'raises error' do
+        expect {get :show, {id: -1}}.to raise_error
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when request is successfully created' do
+      it 'returns 200 status' do
+        create_request_item
+        expect(response).to have_http_status(200)
+      end
+
+      it 'has the correct attributes' do
+        create_request_item
+        request_item = RequestItem.last
+        expect(request_item.request_id).to eq(1)
+        expect(request_item.max_price).to eq(100)
+        expect(request_item.quantity_description).to eq("5")
+        expect(request_item.item.name).to eq("bananas")
+      end
+    end
+
+    context 'when there is a missing parameter' do
+      it 'returns 400 bad request' do
+        create_bad_request_item
+        expect(response).to have_http_status(400)
+      end
+    end
+  end
+
+  describe 'PATCH/PUT #update' do
+    context 'when id does not exist' do
+      it 'raises error' do
+        expect {put '/request_items/-1'}.to raise_error
+      end
+    end
+
+    context 'when a parameter is missing' do
```

Comment:
yeah i guess. doesnt hurt to have it though? idk i was just testing the behavior